pages/treeherder.py

```diff
@@ -59,10 +52,79 @@ def open_perfherder_page(self):
         from perfherder import PerfherderPage
         return PerfherderPage(self.base_url, self.selenium).wait_for_page_to_load()
 
-    def open_single_resultset(self):
-        Wait(self.selenium, self.timeout).until(
-            EC.visibility_of_element_located(self._first_resultset_datestamp_locator))
-        self.selenium.find_element(*self._first_resultset_datestamp_locator).click()
+    def select_next_job(self):
+        el = self.selenium.find_element(*self._result_set_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.ARROW_RIGHT)
+        Wait(self.selenium, self.timeout).until(lambda s: self.result.job_result_status)
+        return self.selenium.find_element(*self._selected_job_title_locator).get_attribute('title')
+
+    class JobDetails(PageRegion):
+
+        _job_details_actionbar_locator = (By.ID, 'job-details-actionbar')
+        _job_details_pin_job_locator = (By.CSS_SELECTOR, '#job-details-actionbar .nav .nav li:nth-child(1) a')
+        _logviewer_button_locator = (By.ID, 'logviewer-btn')
+        _result_status_locator = (By.ID, 'job-details-panel')
+
+        def wait_for_page_to_load(self):
+            body = self.find_element(By.TAG_NAME, 'body')
+            self.wait.until(lambda s: 'loaded' in body.get_attribute('JobDetails'))
+
+        def open_logviewer(self):
+            self.wait_for_page_to_load
```

Comment:
Also, this is a region so we should call this method 'wait_for_region_to_load'. I suspect you perhaps don't even need it as the wait looks like it will never return true, and you're not currently calling it, yet your tests are likely passing?